### PR TITLE
dotty phase 2: scalafix ExplicitNonNullaryApply

### PR DESCRIPTION
**Pull request by ohze**
References #28835, #28837

I split #28837 to some PRs.
This PR:
+ Based on #28947 (dotty phase 1) and contains all commits from that PR.
+ Other changes: Only by running scalafix ExplicitNonNullaryApply
+ Should be reviewed only after #28947 is merged

### CODE REVIEW

**Comment by renatocaval** [comment]
@ohze, I took the liberty to convert this one to a draft because it's waiting for #28947. It makes it more clear that this one is in the frigde for now.

File: akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/javadsl/BehaviorTestKit.scala [hunk 1]
```diff
@@ -75,7 +75,7 @@ abstract class BehaviorTestKit[T] {
   /**
    * The self reference of the actor living inside this testkit.
    */
-  def getRef(): ActorRef[T] = selfInbox.getRef()
+  def getRef(): ActorRef[T] = selfInbox().getRef()
```
Comment (giabao) [inline akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/javadsl/BehaviorTestKit.scala:78]: I prefer changing getRef, selfInbox, TypedProps.apply[T], Props.apply[T] to nullary defs. There are many changes in this PR related to those defs, especial the last one.
But if we do that => break source compatibility 😁

Comment (raboof) [reply akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/javadsl/BehaviorTestKit.scala:78]: in the `javadsl` I don't think it matters much, as it breaks source compatibility only for Scala consumers, not for Java ones, right?

**Review by raboof — APPROVED** [review]
I like the change!

Indeed it does highlight some methods where it is questionable to have the parens, but I don't think that needs to be addressed in this PR.

Methods that I came across where we could consider dropping the parens: FSM `stay`, `ByteStringBuilder.result`, `Supervisor.supervisorStrategy`, `Props.actorProps`, `Props.actorClass`, `LoadMetrics.atFullThrottle`, `ClusterRouterGroup.props`, `TestPublisher.empty`.

**Comment by patriknw** [comment]
> Methods that I came across where we could consider dropping the parens

True, but we can't change that in a source compatible way?

**Review by patriknw — APPROVED** [review]
LGTM, indeed unfortunate that some of these were defined like that originally but not much we can do about that

**Pull request merged**